pkg/lib/indexer/indexer.go

```diff
@@ -283,3 +292,182 @@ func write(dockerfileText, outDockerfile string, logger *logrus.Entry) error {
 
 	return nil
 }
+
+// ExportFromIndexRequest defines the parameters to send to the ExportFromIndex API
+type ExportFromIndexRequest struct {
+	Index         string
+	Operator      string
+	DownloadPath  string
+	ContainerTool string
+}
+
+// ExportFromIndex is an aggregate API used to specify operators from
+// an index image
+func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
+	databaseFile := defaultDatabaseFile
+
+	// set a temp directory
+	workingDir, err := ioutil.TempDir("./", "index_tmp")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(workingDir)
+
+	// Pull the fromIndex
+	if request.Index != "" {
```

Comment:
I think `ExportFromIndex` would be more readable if this block were factored out